test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/initialgainorloss/InitialGainOrLossControllerSpec.scala

```diff
@@ -0,0 +1,272 @@
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.AmountOfMoneyErrorScenarios.amountOfMoneyErrorScenarios
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.onboarding.RedirectToStartBehaviour
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.returns.ReturnsServiceSupport
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.{AuthSupport, ControllerSpec, SessionSupport, returns}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.Generators.{sample, _}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.JourneyStatus.FillingOutReturn
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.finance.AmountInPence
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.InitialGainOrLossAnswers.CompleteInitialGainOrLossAnswers
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.{DisposalDate, DraftReturn}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.{Error, SessionData}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.repos.SessionStore
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.services.returns.ReturnsService
+
+import scala.concurrent.Future
+
+class InitialGainOrLossControllerSpec
+    extends ControllerSpec
+    with AuthSupport
+    with SessionSupport
+    with ReturnsServiceSupport
+    with ScalaCheckDrivenPropertyChecks
+    with RedirectToStartBehaviour {
+
+  implicit lazy val messagesApi: MessagesApi = controller.messagesApi
+
+  override val overrideBindings = List[GuiceableModule](
+    bind[AuthConnector].toInstance(mockAuthConnector),
+    bind[SessionStore].toInstance(mockSessionStore),
+    bind[ReturnsService].toInstance(mockReturnsService)
+  )
+
+  lazy val controller                  = instanceOf[InitialGainOrLossController]
+  implicit lazy val messages: Messages = MessagesImpl(Lang("en"), messagesApi)
+  val initialGainOrLosses              = sample[AmountInPence].copy(value = 2)
+
+  def sessionWithState(
+    answers: CompleteInitialGainOrLossAnswers,
+    disposalDate: DisposalDate
+  ): (SessionData, FillingOutReturn) = {
+    val journey = sample[FillingOutReturn].copy(
+      draftReturn = sample[DraftReturn].copy(
+        initialGainOrLossAnswers = Some(answers)
+      )
+    )
+
+    SessionData.empty.copy(journeyStatus = Some(journey)) -> journey
+  }
+
+  val allQuestionsAnswered: CompleteInitialGainOrLossAnswers = CompleteInitialGainOrLossAnswers(
+    initialGainOrLosses
+  )
+
+  val (session, journey) = sessionWithState(allQuestionsAnswered, sample[DisposalDate])
+  val updatedDraftReturn =
+    journey.draftReturn.copy(initialGainOrLossAnswers = Some(CompleteInitialGainOrLossAnswers(AmountInPence(0))))
+  val updatedSession = session.copy(journeyStatus = Some(journey.copy(draftReturn = updatedDraftReturn)))
+
+  "InitialGainOrLossController" should {
+    "show display" when {
+      def performAction(): Future[Result] = controller.enterInitialGainOrLoss()(FakeRequest())
+      "display a page returning 200" in {
```

Comment:
`InitialGainOrLossController should show display when display a page returning 200` doesn't quite make sense - I'd try and construct a sentence that does

Reply:
you might find a structure like this helpful:
```scala

"InitialGainOrLossController" when {
   "handling requests for x" must {

     "do something" when {

    }

   "do something else" when {
    }
   }

   "handling requests for y" must {

     "do something" when {

    }

   "do something else" when {
    }
   }
}

```
This way the spec is better structured since each endpoint would have it's own section in the spec and you can scope things better so that not all the test data is sitting right at the top where it's far away from where you're using it. There's examples in the other controller specs.

Reply:
Done.